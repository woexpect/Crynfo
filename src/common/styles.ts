import {Dimensions, Platform} from 'react-native';
import {ScaledSheet, verticalScale} from 'react-native-size-matters';

const {width, height} = Dimensions.get('window');

const isIosDevice = Platform.OS === 'ios';

export default ScaledSheet.create({
  // Layout Styles
  completeScreen: {
    flex: 1,
  },
  centerAll: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  completeWidth: {
    width: '100%',
  },
  cryptoListContainer: {
    width: '100%',
    height: verticalScale(64),
  },
  hideOverflow: {
    overflow: 'hidden',
  },
  centerVertically: {
    justifyContent: 'center',
  },
  spaceEvenly: {
    justifyContent: 'space-evenly',
  },
  spaceBetween: {
    justifyContent: 'space-between',
  },
  toEnd: {
    alignItems: 'flex-end',
  },
  roundedTopCorners: {
    borderTopRightRadius: verticalScale(16),
    borderTopLeftRadius: verticalScale(16),
  },
  onTop: {
    position: 'absolute',
  },
  cover: {
    width: '100%',
    height: '100%',
  },
  row: {
    flexDirection: 'row',
  },
  // Paddings
  paddingRight16: {
    paddingRight: verticalScale(16),
  },
  paddingHorizontal: {
    paddingHorizontal: verticalScale(16),
  },
  // Margins
  marginLeft16: {
    marginLeft: verticalScale(16),
  },
  marginRight16: {
    marginRight: verticalScale(16),
  },
  marginTop8: {
    marginTop: verticalScale(8),
  },
  marginTop1: {
    marginTop: 1,
  },
  marginTop16: {
    marginTop: verticalScale(16),
  },
  marginTop24: {
    marginTop: verticalScale(24),
  },
  // Background Colors
  bgBlack: {
    backgroundColor: '#141414',
  },
  bgMain: {
    backgroundColor: '#DEF263',
  },
  // Tint Colors
  tintBlack: {
    tintColor: '#141414',
  },
  tintMain: {
    tintColor: '#DEF263',
  },
  // Text Colors
  tcRed: {
    color: '#F26363',
  },
  tcBlack: {
    color: '#141414',
  },
  tcMain: {
    color: '#DEF263',
  },
  tcBlack8: {
    color: '#1414141F',
  },
  tcBlack64: {
    color: '#141414A3',
  },
  tcMain16: {
    color: '#DEF26329',
  },
  // Border Colors
  bcBlack: {
    borderColor: '#141414',
  },
  bcMain: {
    borderColor: '#DEF263',
  },
  // Font Families
  black: {
    fontFamily: 'Inter-Black',
  },
  bold: {
    fontFamily: 'Inter-Bold',
  },
  semiBold: {
    fontFamily: 'Inter-SemiBold',
  },
  medium: {
    fontFamily: 'Inter-Medium',
  },
  regular: {
    fontFamily: 'Inter-Regular',
  },
  // Font sizes
  cryptoPlaceholder: {
    fontSize: '112@ms0.3',
    includeFontPadding: false,
    lineHeight: '112@ms0.3',
    marginTop: -8,
    marginLeft: -4,
    letterSpacing: -2,
  },
  title: {
    fontSize: '26@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h1: {
    fontSize: '24@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h15: {
    fontSize: '22@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h2: {
    fontSize: '20@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  subtitle: {
    fontSize: '18@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  largerText: {
    fontSize: '16@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  largeText: {
    fontSize: '14@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  mediumText: {
    fontSize: '12@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  normalText: {
    fontSize: '10@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  smallText: {
    fontSize: '6@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
});
