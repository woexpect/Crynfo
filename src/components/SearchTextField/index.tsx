import React from 'react';
import {Image, TextInput, View} from 'react-native';
// Custom components
// User tools
import styles from '../../common/styles';

const SearchTextField = ({
  value,
  onChangeText,
  keyboardType = 'default',
  placeholder = '',
  addStyles,
  secure = false,
  autoCapitalize,
  icon,
  placeholderTextColor,
}: {
  value: any;
  onChangeText: any;
  keyboardType?: any;
  placeholder?: string;
  addStyles?: any;
  secure?: boolean;
  autoCapitalize?: any;
  icon?: any;
  placeholderTextColor?: string;
}) => {
  const setFieldValue = (incomingValue: string) => {
    if (onChangeText) {
      onChangeText(incomingValue);
    }
  };

  return (
    <View style={addStyles}>
      <View style={styles.textFieldContainer}>
        {icon && (
          <View style={styles.iconInputContainer}>
            <Image source={icon} style={[styles.icon, styles.tintMain]} />
          </View>
        )}
        <TextInput
          style={[styles.input, styles.regular, icon && styles.inputWithIcon]}
          onChangeText={setFieldValue}
          value={value}
          placeholder={placeholder}
          placeholderTextColor={placeholderTextColor || '#DEF263'}
          keyboardType={keyboardType}
          secureTextEntry={secure}
          autoCapitalize={autoCapitalize}
          testID="searchTextInput"
        />
      </View>
    </View>
  );
};

export default React.memo(SearchTextField);
