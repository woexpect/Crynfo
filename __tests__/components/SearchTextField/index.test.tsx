import 'react-native';
import React from 'react';
import SearchTextField from '../../../src/components/SearchTextField';

// Note: import explicitly to use the types shiped with jest.
import {expect, it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';

import styles from '../../../src/common/styles';

const searchIcon = require('../../../src/assets/images/icons/search.png');

it('renders correctly with required values only', () => {
  const testValue = 'test';
  const setTestValue = () => {};

  const tree = renderer.create(
    <SearchTextField value={testValue} onChangeText={setTestValue} />,
  );

  expect(tree).toMatchSnapshot();
});

it('renders correctly with all possible values', () => {
  const testValue = 'test';
  const setTestValue = () => {};

  const tree = renderer.create(
    <SearchTextField
      value={testValue}
      onChangeText={setTestValue}
      keyboardType={'number-pad'}
      placeholder="Test Placeholder"
      addStyles={styles.marginTop24}
      secure
      autoCapitalize="none"
      icon={searchIcon}
      placeholderTextColor="#FFFFFF"
    />,
  );

  expect(tree).toMatchSnapshot();
});

it('Should change text value', () => {
  let testValue = 'test';
  const setTestValue = (value: string) => {
    testValue = value;
  };

  const tree = renderer.create(
    <SearchTextField
      value={testValue}
      onChangeText={setTestValue}
      placeholder="Test Placeholder"
    />,
  );

  const searchTextField = tree.root.findByProps({
    placeholder: 'Test Placeholder',
  }).props;

  expect(testValue).toEqual('test');
  act(() => searchTextField.onChangeText('Another thing'));
  expect(testValue).toEqual('Another thing');
});

it('Should run without action set', () => {
  let testValue = 'test';

  const tree = renderer.create(
    <SearchTextField value={testValue} placeholder="Test Placeholder2" />,
  );

  const searchTextField = tree.root.findByProps({
    testID: 'searchTextInput',
  }).props;

  expect(testValue).toEqual('test');
  act(() => searchTextField.onChangeText('Another thing'));
  expect(testValue).toEqual('test');
});
