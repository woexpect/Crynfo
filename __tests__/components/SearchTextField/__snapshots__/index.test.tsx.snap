// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all possible values 1`] = `
<View
  style={
    {
      "marginTop": 47.08235294117647,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#DEF263",
        "borderRadius": 23.541176470588233,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 94.16470588235293,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "height": 47.08235294117647,
          "marginLeft": 34.285714285714285,
          "width": 62.77647058823529,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/images/icons/search.png",
          }
        }
        style={
          [
            {
              "height": 51.42857142857143,
              "width": 51.42857142857143,
            },
            {
              "tintColor": "#DEF263",
            },
          ]
        }
      />
    </View>
    <TextInput
      autoCapitalize="none"
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="Test Placeholder"
      placeholderTextColor="#FFFFFF"
      secureTextEntry={true}
      style={
        [
          {
            "color": "black",
            "flex": 1,
            "height": 94.16470588235293,
            "margin": 0,
            "padding": 0,
            "paddingHorizontal": 21.428571428571427,
          },
          {
            "fontFamily": "Inter-Regular",
          },
          {
            "paddingRight": 85.71428571428571,
          },
        ]
      }
      testID="searchTextInput"
      value="test"
    />
  </View>
</View>
`;

exports[`renders correctly with required values only 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#DEF263",
        "borderRadius": 23.541176470588233,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 94.16470588235293,
        "justifyContent": "space-between",
      }
    }
  >
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder=""
      placeholderTextColor="#DEF263"
      secureTextEntry={false}
      style={
        [
          {
            "color": "black",
            "flex": 1,
            "height": 94.16470588235293,
            "margin": 0,
            "padding": 0,
            "paddingHorizontal": 21.428571428571427,
          },
          {
            "fontFamily": "Inter-Regular",
          },
          undefined,
        ]
      }
      testID="searchTextInput"
      value="test"
    />
  </View>
</View>
`;
